### ISSUE TITLE
If y_continuous_cmap set to True, use continuous cmap in plot

### ISSUE THREAD

**Issue by nightvision04** [erdogant/pca#19]
Could I create a PR for mapping y to a continuous color map? I was thinking syntax could be like this:

` ax = model.biplot3d(n_feat=10, legend=False, y_continuous_cmap=True)`

And it could generate something like the right side (in terms of the point colors).

![image](https://user-images.githubusercontent.com/6148219/131195296-f72f543d-10c1-4882-89a5-388e70e50c2a.png)

**Comment by erdogant** [comment]
Sounds like a great idea! The `cmap` functionality already contains the possibility to reverse the color by adding "_r". My suggestion would be to add "_c" or so for a continuous cmap. What do you think?

**Comment by nightvision04** [comment]
I like your suggestion, however I think I prefer to make this behavior explicit. I'm concerned that we'll need to make a large section in the docstring section to explain each suffix . This might become confusing for some users.

**Comment by erdogant** [comment]
maybe yes. Do you have a suggestion to make the name `y_continuous_cmap` a bit shorter? It is kind of long and attractive for typos. What about: `cmap_gradient `

**Comment by nightvision04** [comment]
I really like `cmap_gradient`. It's elegant. Let's go with that?

**Comment by erdogant** [comment]
👍🏻

**Comment by erdogant** [comment]
Do you have some code on how you would do this?

**Comment by Benfeitas** [comment]
Hi @erdogant , thanks for this. I'm trying to do a continuous gradient between all 3 classes (0, 1 and 2). Slightly adapting your example above I get:

```
import numpy as np
from sklearn.datasets import load_iris

# Load dataset
X = pd.DataFrame(data=load_iris().data, columns=load_iris().feature_names, index=load_iris().target)

# Load pca
from pca import pca
model = pca(n_components=0.95)
model = pca(n_components=3)
results = model.fit_transform(X, verbose=2)
model.scatter(gradient='#5dfa02', cmap='tab20',figsize=(6,6))
```

![image](https://user-images.githubusercontent.com/25486866/208452911-24c5919a-7ad8-425b-a7cb-9958766f632e.png)

I understand that the gradient imposed basically highlights the centroids in each class. However, is it possible to do a continuous gradient between all three classes? An example to highlight the gradient (based on PC1):

```
import matplotlib 

df=results['PC']
fig,ax=plt.subplots()

g = ax.scatter(df['PC1'],df['PC2'], c=df['PC1'], s=30)
mappable = ax.collections[0]
fig.colorbar(mappable=mappable)
```
![image](https://user-images.githubusercontent.com/25486866/208466794-ebe07cd1-407c-4133-b3b7-f1cc50596b76.png)

or based on class label
```
df['class']=df.index
fig,ax=plt.subplots()

g = ax.scatter(df['PC1'],df['PC2'], c=df['class'], s=30)
mappable = ax.collections[0]
fig.colorbar(mappable=mappable)
```
![image](https://user-images.githubusercontent.com/25486866/208467139-b41b1a3a-bb10-465f-9958-5a4707517b5e.png)

**Comment by erdogant** [comment]
I created some changes. Can you update the `scatterd `library and try again?
`pip install -U scatterd
`

`model.scatter(gradient='#5dfa02', cmap='tab20')
`
![image](https://user-images.githubusercontent.com/31008129/208503551-625c31b1-203e-4ddd-9856-047e3d43b616.png)


`model.scatter(gradient='#5dfa02', cmap='tab20', y=np.ones_like(model.results['PC'].index.values))
`
![image](https://user-images.githubusercontent.com/31008129/208503578-0bb1fb5c-7f93-49f7-bcf3-335f65d7b902.png)
